### PR TITLE
fix: Fixing string coercion in tags for catalog

### PR DESCRIPTION
**Pull request by yhakbar**
## Description

Addresses review feedback from #6033.

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [ ] I authored this code entirely myself
- [x] I am submitting code based on open source software (e.g. MIT, MPL-2.0, Apache)
- [x] I am adding or upgrading a dependency or adapted code and confirm it has a compatible open source license
- [x] Update the docs.
- [x] Update the changelog in the docs.
- [x] Run the relevant tests successfully, including pre-commit checks.
- [x] This change is backwards compatible.
- [x] If this change is not forwards compatible (e.g. a new feature), it is gated behind a feature flag.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved component documentation front-matter parsing to correctly handle edge cases including numeric values, boolean-like scalars, and null entries in YAML metadata.

* **Tests**
  * Expanded test coverage for front-matter tag parsing with additional edge case scenarios.

[Review Change Stack](https://app.coderabbit.ai/change-stack/gruntwork-io/terragrunt/pull/6066)

### CODE REVIEW

File: internal/cli/commands/catalog/tui/redesign/component_doc.go [hunk 1]
```diff
@@ -300,36 +315,54 @@ func (doc *ComponentDoc) extractFrontmatter() {
...
+	if node == nil || node.Kind != yaml.ScalarNode || node.Tag == "!!null" {
+		return "", false
+	}
 
-		for _, item := range val {
-			s, ok := item.(string)
-			if !ok {
-				continue
-			}
+	s := strings.TrimSpace(node.Value)
```
Comment (denis256) [inline internal/cli/commands/catalog/tui/redesign/component_doc.go:327]: Interesting, it will work in case of `tags: ["  tomato  "]`

**Review by denis256 — APPROVED** [review]
